### PR TITLE
fix(agw): Converted s11 task files to cpp

### PR DESCRIPTION
**Pull request by VinashakAnkitAman**
## Title
fix(agw): Converted s11 task files to cpp [Extension to Github Issue: #13096]

## Summary
Migrated all the files related to s11 task in the folders `magma/lte/gateway/c/core/oai/tasks/s11/` and `magma/lte/gateway/c/core/oai/include/` to `cpp/hpp` and fixed all the compilation errors and warnings.

Also followed the guidelines for order of headers from https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

## Test plan
Verified compilation with following:
- make clean; make build_oai;
- make clean; bazel build //lte/gateway/c/core:agw_of
~~make clean; bazel build --config=production //lte/gateway/c/core:agw_of~~
- make clean; bazel test //lte/gateway/c/core/...
~~make clean; bazel test //lte/gateway/c/core/oai/test/amf:amf_stateless_test~~
- make clean; make FEATURES=mme_oai build_oai
~~make clean; export FEATURES=mme_oai; make build_oai~~

These compilations are no more valid on latest master after removal of target test_oai from Make file:
```
make clean; make test_oai;
make clean; make test_oai BUILD_TYPE=Debug OAI_TESTS=test_amf_app
```

Verified with sanity and in CI

Signed-off-by: VinashakAnkitAman <ankit.aman@radisys.com>

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/s11/s11_ie_formatter.cpp [hunk 1]
```diff
@@ -948,15 +954,15 @@ status_code_e gtpv2c_failed_bearer_context_within_delete_bearer_request_ie_set(
 //------------------------------------------------------------------------------
 status_code_e gtpv2c_ebis_within_delete_bearer_request_ie_set(
     nw_gtpv2c_msg_handle_t* msg, const ebis_to_be_deleted_t* ebis_tbd) {
-  int rc = RETURNok;
+  nw_rc_t rc = NW_OK;
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/s11/s11_ie_formatter.cpp:957]: Change the prototype

File: lte/gateway/c/core/oai/tasks/s11/s11_ie_formatter.cpp [hunk 2]
```diff
@@ -948,15 +954,15 @@ status_code_e gtpv2c_failed_bearer_context_within_delete_bearer_request_ie_set(
...
   DevAssert(ebis_tbd);
   for (int num_ebi = 0; num_ebi < ebis_tbd->num_ebis; num_ebi++) {
     /** Add EBIs. */
     ebi_t ebi = ebis_tbd->eps_bearer_id[num_ebi];
     rc = nwGtpv2cMsgAddIe(*msg, NW_GTPV2C_IE_EBI, 1, 1, &ebi);
     DevAssert(NW_OK == rc);
   }
-  return NW_OK;
+  return RETURNok;
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/s11/s11_ie_formatter.cpp:965]: parameter, rc and return type needs to be of same data type

**Review by LKreutzer — APPROVED** [review]
Bazel changes lgtm

File: lte/gateway/c/core/oai/tasks/mme_app/mme_app_itti_messaging.cpp [hunk 3]
```diff
@@ -46,7 +46,7 @@ extern "C" {
 #include "lte/gateway/c/core/oai/include/mme_app_desc.hpp"
 #include "lte/gateway/c/core/oai/include/mme_app_ue_context.hpp"
 #include "lte/gateway/c/core/oai/include/mme_config.hpp"
-#include "lte/gateway/c/core/oai/include/s11_messages_types.h"
+#include "lte/gateway/c/core/oai/include/s11_messages_types.hpp"
```
Comment (pruthvihebbani) [inline lte/gateway/c/core/oai/tasks/mme_app/mme_app_itti_messaging.cpp:49]: Move mme_app_itti_messaging.hpp to top

File: lte/gateway/c/core/oai/tasks/s11/s11_mme_bearer_manager.cpp [hunk 4]
```diff
@@ -374,10 +374,9 @@ status_code_e s11_mme_delete_bearer_command(
   }
 
   // Add bearer contexts to be removed.
-  for (int num_ebi = 0; num_ebi < cmd_p->ebi_list.num_ebi; num_ebi++) {
-    rc = gtpv2c_bearer_context_ebi_only_ie_set(&(ulp_req.hMsg),
-                                               cmd_p->ebi_list.ebis[num_ebi]);
-    DevAssert(NW_OK == rc);
+  for (uint32_t num_ebi = 0; num_ebi < cmd_p->ebi_list.num_ebi; num_ebi++) {
```
Comment (pruthvihebbani) [inline lte/gateway/c/core/oai/tasks/s11/s11_mme_bearer_manager.cpp]: uint8_t should be sufficient

**Review by pruthvihebbani — CHANGES_REQUESTED** [review]

File: lte/gateway/c/core/oai/tasks/s11/s11_ie_formatter.cpp (hunk 1 above)
Comment (VinashakAnkitAman) [reply lte/gateway/c/core/oai/tasks/s11/s11_ie_formatter.cpp:957]: The rc is used to get internal function call return values and hence is independent of function return type.

File: lte/gateway/c/core/oai/tasks/s11/s11_ie_formatter.cpp (hunk 2 above)
Comment (VinashakAnkitAman) [reply lte/gateway/c/core/oai/tasks/s11/s11_ie_formatter.cpp:965]: The rc is used to get internal function call return values and hence is independent of function return type.

**Review by pruthvihebbani — APPROVED** [review]
LGTM

File: lte/gateway/c/core/oai/tasks/s11/s11_mme_bearer_manager.cpp [hunk 5]
```diff
@@ -508,9 +507,8 @@ status_code_e s11_mme_create_bearer_response(
                  NwGtpv2c.h etc.. */
 
   for (int i = 0; i < response_p->bearer_contexts.num_bearer_context; i++) {
-    rc = gtpv2c_bearer_context_within_create_bearer_response_ie_set(
+    gtpv2c_bearer_context_within_create_bearer_response_ie_set(
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/s11/s11_mme_bearer_manager.cpp:510]: Instead of assert, we can add log

File: lte/gateway/c/core/oai/tasks/s11/s11_mme_session_manager.cpp [hunk 6]
```diff
@@ -15,57 +15,59 @@
...
     itti_s11_create_session_request_t* req_p) {
-  nw_gtpv2c_ulp_api_t ulp_req = {0};
+  nw_gtpv2c_ulp_api_t ulp_req = {};
   nw_rc_t rc;
   uint8_t restart_counter = 0;
 
   OAILOG_FUNC_IN(LOG_S11);
 
-  if (!stack_p) return RETURNerror;
+  DevAssert(stack_p);
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/s11/s11_mme_session_manager.cpp:70]: Why is this changed?

File: lte/gateway/c/core/oai/tasks/s11/s11_tasks.cpp [hunk 7]
```diff
@@ -356,8 +363,8 @@ static void* s11_mme_thread(void* args) {
 }
 
 //------------------------------------------------------------------------------
-status_code_e s11_mme_init(mme_config_t* mme_config_p) {
-  int ret = 0;
+extern "C" status_code_e s11_mme_init(mme_config_t* mme_config_p) {
+  status_code_e ret = RETURNok;
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/s11/s11_tasks.cpp]: ret is used only once instead we can directly return value

File: lte/gateway/c/core/oai/tasks/s11/s11_mme_session_manager.cpp (hunk 6 above)
Comment (VinashakAnkitAman) [reply lte/gateway/c/core/oai/tasks/s11/s11_mme_session_manager.cpp:70]: This is changed to keep the code consistent with other functions. In other functions, like `s11_mme_handle_create_session_response`,  `s11_mme_delete_session_request`, `s11_mme_handle_delete_session_response` and `s11_mme_handle_ulp_error_indicatior`, same DevAssert is used and failed with assert if stack_p is null.

File: lte/gateway/c/core/oai/tasks/s11/s11_mme_bearer_manager.cpp (hunk 5 above)
Comment (VinashakAnkitAman) [reply lte/gateway/c/core/oai/tasks/s11/s11_mme_bearer_manager.cpp:510]: I did not add the log here because this function call will either assert or return True always and I did not remove the DevAsserts everywhere because these are used everywhere in s11 and gtpv1-u. Removing them should be a separate effort and moreover this code is explicitly not owned by us and in current scope we have the capacity of only validate the compilation and unittests but end to end testing is out of scope.

File: lte/gateway/c/core/oai/tasks/s11/s11_tasks.cpp (hunk 7 above)
Comment (VinashakAnkitAman) [reply lte/gateway/c/core/oai/tasks/s11/s11_tasks.cpp]: Done

**Review by rsarwad — APPROVED** [review]
LGTM
